R/checkmate.R

```diff
@@ -0,0 +1,176 @@
+# assertions ----
+
+#' Additional Assertions for `checkmate`
+#'
+#' @description `r lifecycle::badge("experimental")`
+#'
+#' We provide additional assertion functions that can be used together with
+#' the `checkmate` functions. These are described in individual help pages
+#' linked below.
+#'
+#' @return Depending on the function prefix.
+#' - `assert_` functions return the object invisibly if successful, and otherwise
+#'   throw an error message.
+#' - `check_` functions return `TRUE` if successful, otherwise a string with the
+#'   error message.
+#' - `test_` functions just return `TRUE` or `FALSE`.
+#'
+#' @seealso [assert_probability()], [assert_prob()], [assert_prob_range()].
+#'
+#' @name assertions
+NULL
+
+# assert_probability ----
+
+#' Check if an Argument is a Probability Vector
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' Check if every element in a given numerical vector represent a probability,
+#' that is a number within (0, 1) interval, that can optionally be closed at any
+#' side.
+#'
+#' @param x (`numeric`)\cr vector with numerical values to check.
+#' @param bounds_closed (`logical`)\cr should bounds be closed? This can be a
+#'   scalar or vector of length two. If it is a scalar, then its value applies
+#'   equally to lower bound \eqn{0} and upper bound \eqn{1}. If this is a vector
+#'   with two flags, the first flag corresponds to the lower bound \eqn{0}
+#'   only, and the second to the upper bound \eqn{1} only.
+#' @inheritParams checkmate::check_numeric
+#'
+#' @return `TRUE` if successful, otherwise a string with the error message.
+#'
+#' @seealso [`assertions`] for more details.
+#'
+#' @export
+#' @examples
+#' x <- c(0, 0.2, 0.1, 0.3, 1)
+#' check_probability(x)
+#' check_probability(x, bounds_closed = FALSE)
+#' check_probability(x, bounds_closed = c(FALSE, TRUE))
+check_probability <- function(x, bounds_closed = TRUE, len = NULL, sorted = FALSE) {
```

Comment:
comment on naming: I would propose `check_probabilities` here and `check_probability` below

Reply:
Thank you @danielinteractive, Sure, just one concern I have is that these names are very close to each other and hence easy to make a mistake. I was inspired by `check_integer` / `check_int` when proposed `check_probability` / `check_prob`. Let me know if you are sure about this change, and I will change it then.

Reply:
Ok, so I will change it :)